core/src/test/java/org/infinispan/expiration/impl/ClusterExpirationFunctionalTest.java

```diff
@@ -340,4 +341,49 @@ public void testWriteExpiredEntry() throws InterruptedException {
          ts2.advance(secondOneMilliAdvanced);
       }
    }
+
+   public void testMaxIdleReadNodeDiesPrimary() {
+      testMaxIdleNodeDies(c -> {
+         if (cacheMode.isScattered()) {
+            return new MagicKey(c);
+         }
+         return new MagicKey(c, cache0);
+      });
+   }
+
+   public void testMaxIdleReadNodeDiesBackup() {
+      testMaxIdleNodeDies(c -> {
+         if (cacheMode.isScattered()) {
+            return new MagicKey(cache0);
+         }
+         return new MagicKey(cache0, c);
+      });
+   }
+
+   private void testMaxIdleNodeDies(Function<Cache<?, ?>, MagicKey> keyToUseFunction) {
+      addClusterEnabledCacheManager(TestDataSCI.INSTANCE, configurationBuilder);
+      waitForClusterToForm();
+
+      Cache<Object, String> cache3 = cache(3);
+
+      ControlledTimeService ts4 = new ControlledTimeService();
+      TestingUtil.replaceComponent(manager(3), TimeService.class, ts4, true);
+
+      MagicKey key = keyToUseFunction.apply(cache3);
+
+      // We always write to cache3 so that scattered uses it as a backup if the key isn't owned by it
+      cache3.put(key, "max-idle", -1, TimeUnit.MILLISECONDS, 100, TimeUnit.MILLISECONDS);
+
+      long justbeforeExpiration = 99;
+      incrementAllTimeServices(justbeforeExpiration, TimeUnit.MILLISECONDS);
+      ts4.advance(justbeforeExpiration);
+
+      assertNotNull(cache3.get(key));
+
+      killMember(3);
```

Comment:
`killMember(3)` calls `waitForNoRebalance(caches)`, I think you need to block the rebalance and test expiration during the rebalance as well.

Reply:
Yeah, this test wasn't trying to test that out yet. It was more to make sure after node failure that the timestamp is still good.

Reply:
@danberindei is there an example test I can use that you recommend that blocks rebalance? I find getting a test correct with that can be quite annoying.

Reply:
There are many tests that block a rebalance, in different ways, depending on the scenario they want to test.

Some block the `StateResponseCommand` so that the they block the insertion of particular keys. Some only block the `CacheTopologyControlCommand(REBALANCE_PHASE_CONFIRM)` commands so the cache stays in a particular topology phase (usually `READ_OLD_WRITE_ALL`) after state transfer has finished.

Some block the sender side, with `ControlledRpcManager` or `ControlledTransport`. Some block the receiver side, with many impls of `InboundInvocationHandler`, `PerCacheInboundInvocationHandler`, `LocalTopologyManagerImpl`, `StateProviderImpl`, or `StateConsumerImpl`.